### PR TITLE
chore: warn on unknown mode

### PR DESCRIPTION
**Pull request by TheAlexLichter**
Warn if mode is neither `spa` nor `universal`.

## Types of changes
- [x] Warning

### CODE REVIEW

File: packages/cli/src/utils/index.js [hunk 1]
```diff
@@ -56,6 +56,10 @@ export async function loadNuxtConfig(argv) {
   options.mode =
     (argv.spa && 'spa') || (argv.universal && 'universal') || options.mode
 
+  if (options.mode && !['spa', 'universal'].includes(options.mode)) {
```
Comment (pi0) [inline packages/cli/src/utils/index.js]: The way that `mode` is designed is like a defined preset. We can add more modes using `options.modes`.  So i suggest moving this check after resolving `modePreset`: https://github.com/nuxt/nuxt.js/blob/dev/packages/config/src/options.js#L240

Comment (manniL) [reply packages/cli/src/utils/index.js]: I felt that I missed something like that. Thanks for the heads up! 🙈

**Review by aldarund — APPROVED** [review]
nice idea. recently in discord #russian channel someone asked that he used mode: 'static' and was wondering why it dont generate static html files :)

**Review by pi0 — APPROVED** [review]

**Comment by AndrewBogdanovTSS** [comment]
@aldarund I was also caught in this pitfall when I was trying out new `modern` mode. For some reason I decided that it should be specified like `mode: 'modern'` and since I didn't get any warnings/nor errors during the build - I was convinced that it's the right way to go, although nothing been working in the browser. It just was an endless wait for the response

**Pull request merged**